### PR TITLE
Implement noshow result

### LINKED ISSUE #27: Allow players to report no-shows.
This should probably just be a flag in the Match Listings, and not a fully automated thing.

But it's better than reporting 2-0 with no additional details.

### PR DESCRIPTION
**Pull request by stash86**
Add extra codes to diferentiate normal win and noshow win, and let player to report noshow

Fixes #27

### CODE REVIEW

File: deck.php [hunk 1]
```diff
@@ -510,11 +510,11 @@ function matchupTable($deck)
             }
             if ($match->playerWon($deck->playername)) {
                 $color = '#009900';
-                $res = 'Win';
+                $res = 'Win'.($match->result == 'WO' ? ' Noshow' : '');
```
Comment (silasary) [inline deck.php]: Can we use `NS` rather than WO?

File: event.php [hunk 2]
```diff
@@ -809,7 +809,11 @@ function printUnverifiedPlayerCell($match, $playername)
     }
     if (($match->getPlayerWins($playername) > 0) || ($match->getPlayerLosses($playername) > 0)) {
         if ($match->getPlayerWins($playername) > $match->getPlayerLosses($playername)) {
-            $matchresult = 'W ';
+            if ($match->result == 'WO') {
+                $matchresult = '<span style="font-size: 14pt" title ="Opponent doesn\'t show up">&#x26A0;</span> W ';
```
Comment (silasary) [inline event.php]: "Opponent didn't show up"

**Review by silasary — CHANGES_REQUESTED** [review]

**Comment by silasary** [comment]
I'm also concerned that this will give TOs the impression that Gatherling will automatically handle NoShows, when in fact they still require manual intervention.

**Comment by stash86** [comment]
Well, TO does have to manually enters the "NS" result anyway. We can add automatic drop in the code for NS player, which only fires after TO input the result from Host CP

File: deck.php (hunk 1 above)
Comment (stash86) [reply deck.php]: Yes we can

**Comment by stash86** [comment]
Changed all 'WO' into 'NS'. Now I'm gonna add auto-drop for NS player, unless we are afraid of any player abuse (which I personally think there is no hole to be abused)

**Comment by silasary** [comment]
I don't want auto-drop.  We have had cases of people accidentally reporting No-Shows early, and then their opponent showing up.

And that's not even considering the case of someone getting salty and dropping their opponent out of spite.

**Comment by stash86** [comment]
So let's say player A reports their match as a NS win. Based on my proposed code, there are 4 cases that can happens (I haven't work on the auto-drop code yet)

1. Player B later on reports their side of result as a 0-2 loss. The NS tag will be removed. (The auto-drop function won't be called)

2. Player A and player B later on play their match. And then Player B reports their side of actual result. When the result conflicted, the code will ask both players to enter the correct result. Once it happens, the NS tag will be removed. (The auto-drop function won't be called)

3. The host pushed the result through Event CP, but they choose another result than the NS Win. The NS tag will be removed. (The auto-drop function won't be called)

4. The host pushed the result through Event CP, and they reports it as a NS 2-0 win for Player A. (The auto-drop function will be called)

**Pull request closed without merging**